test_runner/src/main/kotlin/ftl/mock/MockServer.kt

```diff
@@ -180,6 +190,35 @@ object MockServer {
                     val projectId = call.parameters["project"]
 
                     val matrixId = matrixIdCounter.incrementAndGet().toString()
+                    val requestBody = withContext(Dispatchers.IO) {
+                        GZIPInputStream(call.receive<ByteArray>().inputStream()).bufferedReader(UTF_8).use {
+                            ObjectMapper().readValue<Map<String, Any>>(it.readText())
+                        }
+                    }
+                    val clientName = requestBody["clientInfo"]?.objectToMap()?.get("name") as String
+                    val allClientDetails = mutableMapOf<String, String>()
+                    requestBody["clientInfo"]
+                        ?.objectToMap()
+                        ?.get("clientInfoDetails")?.let { list ->
+                            if (list is List<*>) {
+                                list.forEach { map ->
+                                    if (map is Map<*, *>) {
```

Comment:
instead of `if (map is Map<*, *>)` I would use `filter {it is Map}` or `mapNotNull{it as? Map}`